float32 Vref
float32 Psiref